### ISSUE TITLE
Malformed after-Interceptor may lead to runtime errors

### ISSUE THREAD

**Issue by jean-bernard-valentaten** [dotmailer/dotmailer-magento2-extension-b2b#1]
In the class `\Dotdigitalgroup\B2b\Plugin\NegotiableQuotePlugin`, the `afterSave` method returns `void` although the observed method `\Magento\NegotiableQuote\Model\NegotiableQuoteRepository::save()` returns a boolean.
This may lead to issues when other `after` interceptors expect the `$result` to be a boolean.

Unfortunately Magento (as of 2.3.5-p2) does not yet enforce the return value by adding a return type to the observed method's signature, Magento's implementation does return a boolean in all cases.

I therefore propose the following patch to fix this issue:
```patch
diff --git a/Plugin/NegotiableQuotePlugin.php b/Plugin/NegotiableQuotePlugin.php
--- a/Plugin/NegotiableQuotePlugin.php
+++ b/Plugin/NegotiableQuotePlugin.php
@@ -42,7 +42,7 @@
 
         //If data already exists in quoteModel we do not need to create new record in email_b2b_quote table
         if ($this->negotiableQuoteRepository->getByQuoteId($quoteModel->getQuoteId())->getData()) {
-            return;
+            return $result;
         }
 
         $customerObject = $this->customerRepositoryInterface->getById($quoteModel->getCreatorId());
@@ -53,6 +53,7 @@
             ->setWebsiteId($customerObject->getWebsiteId());
 
         $this->negotiableQuoteRepository->save($ddgQuote);
+        return $result;
     }
 
     /**

```

**Comment by sta1r** [comment]
@jean-bernard-valentaten Apologies for not responding sooner. You're absolutely right of course, the plugin should return the `$result`. We'll add the change in our next release.

**Comment by sta1r** [comment]
This will be released in our 4.10.0 release, with Magento 2.4.2. Thanks again for raising.
